Platform/ApollolakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c

```diff
@@ -97,6 +100,163 @@ CreateNhltAcpiTable (
   return PlatformService->AcpiTableUpdate ((UINT8 *)Table, Table->Header.Length);
 }
 
+/**
+  Add a Smbios type string into a buffer
+
+**/
+STATIC
+EFI_STATUS
+AddSmbiosTypeString (
+  SMBIOS_TYPE_STRINGS  *Dest,
+  UINT8                 Type,
+  UINT8                 Index,
+  CHAR8                *String
+  )
+{
+  UINTN   Length;
+
+  Dest->Type    = Type;
+  Dest->Idx     = Index;
+  if (String != NULL) {
+    Length = AsciiStrLen (String);
+
+    Dest->String  = (CHAR8 *)AllocateZeroPool (Length + 1);
+    if (Dest->String == NULL) {
+      return EFI_OUT_OF_RESOURCES;
+    }
+    CopyMem (Dest->String, String, Length);
+  }
+
+  return EFI_SUCCESS;
+}
+
+/**
+  Initialize necessary information for Smbios
+
+  @retval EFI_SUCCESS             Initialized necessary information successfully
+  @retval EFI_OUT_OF_RESOURCES    Failed to allocate memory for Smbios info
+
+**/
+EFI_STATUS
+InitializeSmbiosInfo (
+  VOID
+  )
+{
+  CHAR8                 TempStrBuf[SMBIOS_STRING_MAX_LENGTH];
+  UINT16                Index;
+  UINT16                PlatformId;
+  UINTN                 Length;
+  SMBIOS_TYPE_STRINGS  *TempSmbiosStrTbl;
+  BOOT_LOADER_VERSION  *VerInfoTbl;
+  VOID                 *SmbiosStringsPtr;
+
+  Index         = 0;
+  PlatformId    = GetPlatformId ();
+  TempSmbiosStrTbl  = (SMBIOS_TYPE_STRINGS *) AllocateTemporaryMemory (0);
+  VerInfoTbl    = GetLoaderGlobalDataPointer()->VerInfoPtr;
+
+  //
+  // SMBIOS_TYPE_BIOS_INFORMATION
+  //
+  AddSmbiosTypeString (&TempSmbiosStrTbl[Index++], SMBIOS_TYPE_BIOS_INFORMATION,
+    1, "Intel Corporation");
+  if (VerInfoTbl != NULL) {
+    AsciiSPrint (TempStrBuf, sizeof (TempStrBuf),
+      "SBL:%03d.%03d.%03d.%03d.%03d.%05d.%c-%016lX%a\0",
+      VerInfoTbl->ImageVersion.SecureVerNum,
+      VerInfoTbl->ImageVersion.CoreMajorVersion,
+      VerInfoTbl->ImageVersion.CoreMinorVersion,
+      VerInfoTbl->ImageVersion.ProjMajorVersion,
+      VerInfoTbl->ImageVersion.ProjMinorVersion,
+      VerInfoTbl->ImageVersion.BuildNumber,
+      VerInfoTbl->ImageVersion.BldDebug ? 'D' : 'R',
+      VerInfoTbl->SourceVersion,
+      VerInfoTbl->ImageVersion.Dirty ? "-dirty" : "");
+  } else {
+    AsciiSPrint (TempStrBuf, sizeof (TempStrBuf), "%a\0", "Unknown");
+  }
+  AddSmbiosTypeString (&TempSmbiosStrTbl[Index++], SMBIOS_TYPE_BIOS_INFORMATION,
+    2, TempStrBuf);
+  AddSmbiosTypeString (&TempSmbiosStrTbl[Index++], SMBIOS_TYPE_BIOS_INFORMATION,
+    3, "Unknown date");
+
+  //
+  // SMBIOS_TYPE_SYSTEM_INFORMATION
+  //
+  AddSmbiosTypeString (&TempSmbiosStrTbl[Index++], SMBIOS_TYPE_SYSTEM_INFORMATION,
+    1, "Intel Corporation");
+  if (PlatformId == PLATFORM_ID_OXH) {
+    AsciiSPrint (TempStrBuf, sizeof (TempStrBuf), "%a\0", "OXH Client Platform");

```

Comment:
Let us use full name:  Leaf Hill CRB, Oxbow Hill CRB, Juniper Hill CRB, UP Squared Board, Gordon Peak MRB, MinnowBoad 3

Reply:
I have changed it but I don't know if it is what you expected.